src/ComponentRegister.test.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import ComponentRegister from "./ComponentRegister";
+
+import { create } from "react-test-renderer";
+
+beforeEach(() => {
+  ComponentRegister.Link = "a";
+});
+
+it("Registers a link component provided", () => {
+  ComponentRegister.init({
+    // eslint-disable-next-line react/display-name, react/prop-types
+    Link: ({ href, children }) => <div data-href={href}>{children}</div>
+  });
+
+  const component = create(
+    <ComponentRegister.Link href="/"> Test </ComponentRegister.Link>
+  );
+  expect(component).toMatchSnapshot();
+});
+
+it("Registers an anchor tag if not initialised", () => {
+  const component = create(
+    <ComponentRegister.Link href="/"> Test </ComponentRegister.Link>
```

Comment:
Is there a particular reason you put spaces in these? It looks a bit odd to me (and will render the spaces for some tags).